certificate-manager/src/test/java/io/strimzi/certs/IpAndDnsValidationTest.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.certs;
+
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class IpAndDnsValidationTest {
+    @Test
+    public void testIPv6()   {
+        assertThat(IpAndDnsValidation.isValidIpv6Address("::1"), is(true));
+        assertThat(IpAndDnsValidation.isValidIpv6Address("fc01::8d1c"), is(true));
+        assertThat(IpAndDnsValidation.isValidIpv6Address("::fc01:8d1c"), is(true));
+        assertThat(IpAndDnsValidation.isValidIpv6Address("fc01:8d1c::"), is(true));
+        assertThat(IpAndDnsValidation.isValidIpv6Address("1762:0:0:0:0:B03:1:AF18"), is(true));
+
+        assertThat(IpAndDnsValidation.isValidIpv6Address("fc01::8j1c"), is(false));
+        assertThat(IpAndDnsValidation.isValidIpv6Address("fc01::176j::8j1c"), is(false));
```

Comment:
You tested a non-hex char above. I think here you want to test with valid hex chars only and that > 1 `::` is rejected.